src/main/python/deep_qa/layers/attention/matrix_attention.py

```diff
@@ -39,7 +41,8 @@ def __init__(self, **kwargs):
         self.similarity_function = similarity_functions[sim_function_choice](**similarity_function_params)
 
     def build(self, input_shape):
-        self.trainable_weights = self.similarity_function.initialize_weights(input_shape)
+        similarity_function_shape = self.get_output_shape_for(input_shape) + (input_shape[0][-1],)
```

Comment:
`input_shape` is `[(batch_size, num_rows_1, embedding_dim), (batch_size, num_rows_2, embedding_dim)]`.  That's not the shape that `self.similarity_function.initialize_weights` is expecting.  This is just fixing a bug.